samtranslator/model/sam_resources.py

```diff
@@ -1310,6 +1311,7 @@ class SamApi(SamResourceMacro):
         "Stage": ApiGatewayStage.resource_type,
         "Deployment": ApiGatewayDeployment.resource_type,
         "DomainName": ApiGatewayDomainName.resource_type,
+        "DomainNameV2": ApiGatewayDomainNameV2.resource_type,
```

Comment:
Do you know what this does? `referable_properties` sounds like things you can `!Ref`. and I don't know if this means we need to do something else for this to actually work, or if it just does.

Reply:
I don't think there's anything else that needs to be done for this to work. We can ref it using <api-logicalId>.DomainNameV2 property. https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-specification-generated-resources-api.html